index.js

```diff
@@ -136,8 +136,9 @@ function renderTag(elem, opts) {
   }
 
   if (
-    opts.xmlMode
-    && (!elem.children || elem.children.length === 0)
+    (opts.xmlMode
+    && (!elem.children || elem.children.length === 0)) ||
```

Comment:
Ideally only self closing tags which are listed under `singleTag` should have self closing. Moving the check along with `!elem.children || elem.children.length === 0` will make any element which has no children to be self closing. Example
`<iframe src="test"></iframe>` will become `<iframe src="test"/>`